jaxopt/_src/zoom_linesearch.py

```diff
@@ -138,7 +150,7 @@ class ZoomLineSearchState(NamedTuple):
 
   error: float
   done: bool
-  fail_code: int  # encode failure status, see _check_status
+  error_flag: int  # encode failure status, see _check_status
```

Comment:
What's the motivation for changing the name? Since, other solvers use `fail_code`, this would create an inconsistency.

Reply:
The NaN/Inf flag does not necessarily lead to a failure but was more of a warning. One could a priori fully remove it now too. Other than that, we can use fail_code. 
PS: Which other algorithms use fail_code? I could further harmonize if you want.

Reply:
But in that case error != warning too. `fail_code` is only used in the zoom linesearch but `failed_linesearch` is used in many places so I would prefer to keep `fail_code` for consistency.

Reply:
Ok! The changes have been made, I've squashed everything.